obspy/clients/fdsn/mass_downloader/download_helpers.py

```
@@ -0,0 +1,1140 @@
+                        utils.safe_delete(interval.filename)
+                        discarded_bytes += size
+                        interval.status = STATUS.DOWNLOAD_REJECTED
+                        continue
+
+                    if self.restrictions.minimum_length:
+                        duration = sum([tr.stats.endtime - tr.stats.starttime
+                                        for tr in st])
+                        expected_min_duration = \
+                            self.restrictions.minimum_length * \
+                            (interval.end - interval.start)
+                        if duration < expected_min_duration:
+                            self.logger.info(
+                                "File '%s' has only %.2f seconds of data. "
+                                "%.2f are required. File will be deleted." %
+                                (interval.filename, duration,
+                                 expected_min_duration))
+                            utils.safe_delete(interval.filename)
+                            discarded_bytes += size
+                            interval.status = STATUS.DOWNLOAD_REJECTED
+                            continue
+
+                    downloaded_bytes += size
+                    interval.status = STATUS.DOWNLOADED
+        return downloaded_bytes, discarded_bytes
+
+    def _parse_miniseed_filenames(self, filenames, restrictions):
+        time_range = restrictions.minimum_length * (restrictions.endtime -
+                                                    restrictions.starttime)
+        channel_availability = []
+        for filename in filenames:
+            st = obspy.read(filename, format="MSEED", headonly=True)
+            if restrictions.reject_channels_with_gaps and len(st) > 1:
+                self.logger.warning("Channel %s has gap or overlap. Will be "
+                                    "removed." % st[0].id)
+                try:
+                    os.remove(filename)
```

OK, now I see it doesn't ignore errors, so carry on then...